src/relentless/model/potential/potential.py

```diff
@@ -490,3 +490,293 @@ def _derivative(self, param, r, **params):
 
         """
         pass
+
+
+class BondedSpline(BondedPotential):
+    """Base class for bonded spline potentials.
+
+    The bonded spline potential is defined by interpolation through a set of
+    knot points. The interpolation scheme uses Akima splines.
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    num_knots : int
+        Number of knots.
+    mode : str
+        Mode for storing the values of the knots in
+        :class:`~relentless.variable.Variable` that can be optimized. If
+        ``mode='value'``, the knot amplitudes are manipulated directly.
+        If ``mode='diff'``, the amplitude of the *last* knot is fixed, and
+        differences between neighboring knots are manipulated for all other
+        knots. Defaults to ``'diff'``.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Examples
+    --------
+    The spline potential is setup from a tabulated potential instead of
+    specifying knot parameters directly::
+
+        spline = relentless.potential.bond.BondSpline(types=("bondA",), num_knots=3)
+        spline.from_array(("bondA"),[0,1,2],[4,2,0])
+
+    However, the knot variables can be iterated over and manipulated directly::
+
+        for r,k in spline.knots("bondA"):
+            k.value = 1.0
```

Comment:
How would you hand this example given that this is now the base class?

Reply:
The derived classes actually still need a docstring, so you can put specific examples using the right class in each of those. This class should probably get a note saying that it should not be instantiated directly, rather than an example, or else this example should be generic and use "A" as the type.

Small thing to note above: it should be `potential.BondedSpline`